### PR TITLE
Finish auditing the existing error messages

### LINKED ISSUE #295: Audit existing error messages to improve helpfulness
As part of improving error messages #294, we need to audit the existing error messages and improve their helpfulness. Currently, many of them are simply writing the underlying error message out to the console, which is often useless to the end user. We should add the context of what Notion was trying to do when the error occurred, and if possible, what the user can do to fix the problem on their own.

### PR DESCRIPTION
**Pull request by charlespierce**
Closes #295 

Info
-----
* Builds on #328 
* Completes audit and update of existing error messages.
* All error messages in `error/details.rs` should now follow the new style or have a comment explaining why they don't have a CTA.

Changes
-----
* Updated all remaining errors in `error/details.rs` to reflect the new style, mostly by adding CTAs.
* Updated call sites where necessary to pass the correct context to the errors.
* Fixed tests to reflect the new error messages.

### CODE REVIEW

File: crates/notion-core/src/error/details.rs [hunk 1]
```diff
@@ -343,10 +372,13 @@ Please ensure you have correct permissions to the Notion directory."#, name),
...
-            ErrorDetails::ToolNotImplemented => write!(f, "this tool is not yet implemented"),
-            ErrorDetails::UnrecognizedShell { name } => write!(f, "Unrecognized shell: {}", name),
+            ErrorDetails::UnrecognizedShell { name } => write!(f, "Unrecognized shell '{}'
+
+Please ensure you are using a supported shell.", name),
             ErrorDetails::UnspecifiedPostscript => {
-                write!(f, "Notion postscript file not specified")
+                write!(f, "Could not determine Notion postscript file.
+
+Please nesure Notion was installed correctly.")
```
Comment (chriskrycho) [inline crates/notion-core/src/error/details.rs]: Typo: "nesure" → "ensure"

File: crates/notion-core/src/error/details.rs [hunk 2]
```diff
@@ -184,26 +186,39 @@ impl fmt::Display for ErrorDetails {
...
-                version,
+                existing_package,
+                new_package,
             } => write!(
                 f,
-                "Conflict with bin '{}' already installed by '{}' version {}",
-                bin_name, package, version
+                "Executable '{}' is already installed by {}
+
+Remove {} before installing {}",
```
Comment (chriskrycho) [inline crates/notion-core/src/error/details.rs]: For consistency with the other CTAs, this should probably be `Please remove`.

File: crates/notion-core/src/error/details.rs (hunk 1 above)
Comment (charlespierce) [reply crates/notion-core/src/error/details.rs]: Good catch, fixed!

File: crates/notion-core/src/error/details.rs (hunk 2 above)
Comment (charlespierce) [reply crates/notion-core/src/error/details.rs]: Very true, updated.

**Pull request merged**